src/txkube/test/test_authentication.py

```diff
@@ -68,11 +77,50 @@
 -----END CERTIFICATE-----
 """
 
+# Let hostname u"example.invalid" map to an
+# IPv4 address in the TEST-NET range.
+HOST_MAP = {
+    u"example.invalid.": "192.0.2.2"
+}
+
+def createReactor():
+    """
+    Twisted 17.1.0 and higher requires a reactor which implements
+    ``IReactorPluggableNameResolver``.
+    """
+
+    @implementer(IHostResolution)
+    @attr.s
+    class Resolution(object):
+        name = attr.ib()
+
+    class _FakeResolver(object):
+
+        def resolveHostName(self, resolutionReceiver, hostName, *args,  **kwargs):
+            portNumber = kwargs.pop('portNumber')
+            r = Resolution(name=hostName)
+
+            resolutionReceiver.resolutionBegan(r)
+            if hostName in HOST_MAP:
+                resolutionReceiver.addressResolved(
+                    IPv4Address('TCP', HOST_MAP[hostName], portNumber))
+            resolutionReceiver.resolutionComplete()
+            return r
+
+    @implementer(IReactorPluggableNameResolver)
+    class _ResolvingMemoryClockReactor(MemoryReactorClock):
+        nameResolver = _FakeResolver()
+
+    return _ResolvingMemoryClockReactor()
+
+
+
 class AuthenticateWithServiceAccountTests(TestCase):
     """
     Tests for ``authenticate_with_serviceaccount``.
     """
-    def _authorized_request(self, token, headers):
+    def _authorized_request(self, token, headers,
+                            kubernetesHost=b"example.invalid."):
```

Comment:
I guess this should be `kubernetes_host`.

Reply:
OK